### PR TITLE
Basic DatatbricksCluster Class

### PR DESCRIPTION
**Pull request by benrutter**
Hey @jacobtomlinson - not sure if your taking contributions on this so early, but I'm pretty excited about the project and wanted to give it a spin for how it might play out. Put some stuff together here to roughly match the API you specified in [the initial scope](https://github.com/jacobtomlinson/dask-databricks/issues/1)

I've put together some boilerplate (with the helm cluster stuff as a basis). It's just a wrapper for a localcluster but making sure it matches the setup from CLI. As a starter, this is more or less just a wrapper for specifying ports / IP address as done in [the initial github chat](https://github.com/jacobtomlinson/dask-databricks/issues/3)

Giving this a quick run on databricks shows it seems to work as expected, so with this + the initial CLI there'd a very basic set up to get up and running with dask on databricks without the need for custom code.

I've put some very limited testing in place - although I'm not sure what test running you're planning on so the import structure might be  out (I tend to use ```python -m pytest``` on projects but I think dask testing normally conforms to ```py.test```).

Anyway, hope this helps but I know this project is still in very early stages- I won't be touchy if you already have a plan and would rather just outright reject this PR 😁

### CODE REVIEW

File: dask_databricks/databricks/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .databrickscluster import DatabricksCluster
```
Comment (jacobtomlinson) [inline dask_databricks/databricks/__init__.py:1]: I'm curious about the `dask_databricks.databricks` submodule. I feel this adds unnecessary nesting. Maybe we could move this whole thing up a level and just have `dask_databricks/databrickscluster.py`?

File: dask_databricks/databricks/databrickscluster.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import os
+
+from distributed.deploy import LocalCluster
+from distributed.core import rpc
+
+class DatabricksCluster(LocalCluster):
```
Comment (jacobtomlinson) [inline dask_databricks/databricks/databrickscluster.py:6]: We should probably use `Cluster` as a base, `LocalCluster` has a lot of unnecessary `SpecCluster` related baggage attached. However, there is a lot of history around these base classes and I wouldn't want to suggest that you implement this as it is not a fun task.

I think this is a good opportunity for me to clean some things up in `distributed` and then switch this over.

File: dask_databricks/databricks/tests/test_databricks.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import os
+
+import pytest
```
Comment (jacobtomlinson) [inline dask_databricks/databricks/tests/test_databricks.py:3]: Yay for `pytest`. We use it throughout Dask so this is a good choice. We haven't set up CI yet but that's fine.

**Review by jacobtomlinson — APPROVED** [review]
Thank you so much for diving in here @benrutter! As you say it's pretty early and things are moving quickly, but the more the merrier!

I have a lot of thoughts about this (I've left a few comments), and some context that would lead me to make slightly different decisions. But I think this is a good opportunity to say "yes and ..." rather than "no but...".

So I'm going to hit merge and then start iterating on this.

File: dask_databricks/databricks/__init__.py (hunk 1 above)
Comment (benrutter) [reply dask_databricks/databricks/__init__.py:1]: I agree! I basically just copied the structure from helm cluster but that is a lot more complex and sits within dask-kubernetes. The whole thing is under 50 lines atm and if it grows in complexity, moving it into a submodule is probably a fairly easy thing to do.

File: dask_databricks/databricks/databrickscluster.py (hunk 2 above)
Comment (benrutter) [reply dask_databricks/databricks/databrickscluster.py:6]: I should probably admit that I don't understand very much at all about the base classes here 😅 LocalCluster seemed a really quick way to get it up and running whereas Cluster looked like I had to implement a lot of additional stuff to get it going. I assume cluster would give more ability to expand out with stuff like dashboarding and autoscale requests etc though.

**Pull request merged**